### ISSUE TITLE
[Suggestion] Restrict inputs from keyboard or mouse to the game's window only

### ISSUE THREAD

**Issue by TeddyHartanto** [SerpentAI/SerpentAI#134]
* **Operating System:** Linux
* **Serpent Version:** 0.1.12b1
* **Game:** Any game. But I was playing with [gnome-mines](https://wiki.gnome.org/Apps/Mines)
* **Backend:** CPU

### Expected result
Any keyboard or mouse input should be restricted to the game's window.

### Encountered result
The `GameAgent` does mouse clicks or keyboard presses even outside of the game's window.  
See it in action [here](https://www.twitch.tv/videos/137509018) 

### Steps to reproduce

1. Code a mouse click or keyboard press in your `GameAgent`'s `handle_play()`
2. Launch the game and your game agent.

### Remarks
I think it's kinda annoying that the game agent sends inputs on unintended windows. Nasty things can happen, as can be seen from your Twitch video. I wonder if you have researched into this matter? Is it possible to restrict the inputs to only the game's window?

**Comment by nbrochu** [comment]
It's is currently restricted to the game window having focus. The second the window focus is gone, the agent stops receiving frames which should stop your frame handler from doing anything.

Did you encounter something like this yourself or are you just reacting to the Twitch clip? The framework being built while live means a lot of weird/funny/unexpected stuff can happen and people like to clip those for entertainment. This specific video is 9 months old. I obviously ended up adding some preventive measures after that situation happened ;)

**Comment by TeddyHartanto** [comment]
I encountered it while following your [Hello World wiki](https://github.com/SerpentAI/SerpentAI/wiki/The-Serpent.AI-Hello-World) and tweaking the `handle_play()` to simulate a mouse click every time it fires.

Actually, you're right that once the game window focus is gone, the agent stopped sending inputs! At first, I got the wrong impression that it's still sending inputs because after I did `serpent play` on my terminal, the agent fires, and my mouse starts selecting the texts in the terminal. Now that I tested it again, I just realized that it clicks only once. Then the game window is out of focus and it no longer sends any input. Sorry for the false alarm! :stuck_out_tongue: Closing this issue!
